### PR TITLE
Add a couple extra fields for caption and reaction to shared posts/stories/reels

### PR DESCRIPTION
**Pull request by vurpo**

### CODE REVIEW

**Comment by vurpo** [comment]
This should finally contain the fields we discussed with Bradley yesterday that are needed, and works in my experiments at least

File: mautrix_instagram/portal.py [hunk 1]
```diff
@@ -1192,6 +1191,10 @@ async def _convert_instagram_media_share(
                 }
                 combined["formatted_body"] = combined_formatted_body
 
+            if share_item.caption and item_type_name != "clip":
```
Comment (sumnerevans) [inline mautrix_instagram/portal.py:1194]: why restrict by `item_type_name`? I feel like we should just include it all the time, even if it doesn't get used by the clients

File: mautrix_instagram/portal.py [hunk 2]
```diff
@@ -1263,6 +1267,8 @@ async def _convert_instagram_xma_media_share(
                 f"<strong>{escaped_header_text}</strong>"
                 f"{escaped_caption_text[len(escaped_header_text):]}"
             )
+            content["com.beeper.raw_caption_text"] = caption_text[len(header_text) :]
+        content["com.beeper.instagram_item_username"] = media.header_title_text
```
Comment (sumnerevans) [inline mautrix_instagram/portal.py:1271]: are you sure that this is just the username and not any additional text?

Comment (vurpo) [reply mautrix_instagram/portal.py]: Yeah, I'll move it into the if statement where we're already assuming that is the username

**Review by sumnerevans — APPROVED** [review]
Looks like it's probably fine. We can always modify if there's edge cases we haven't covered.

**Pull request merged**